src/header.iced

```diff
@@ -0,0 +1,101 @@
+{prng} = require('crypto')
+{createHash} = require('crypto')
+msgpack = require('msgpack-lite')
+crypto = require('keybase-nacl')
+nonce = require('./nonce.iced')
+util = require('./util.iced')
+
+encryption_mode = 0
+attached_sign_mode = 1
+detached_sign_mode = 2
+current_major = 1
+current_minor = 0
+crypto_auth_BYTES = 32
+crypto_secretkey_BYTES = 32
+
+compute_mac_key = (encryptor, header_hash, pubkey) ->
+  zero_bytes = Buffer.alloc(32)
+  mac_box = encryptor.encrypt({plaintext : zero_bytes, nonce : nonce.nonceForMACKeyBox(header_hash), pubkey})
+  # take last crypto_auth_BYTES bytes of MAC box
+  return mac_box.slice(-crypto_auth_BYTES)
```

Comment:
It so happens that `BYTES` (the output length) and `KEYBYTES` (the key length) are equal for `crypto_auth`, but I think it's technically `KEYBYTES` that we want to depend on here, since we're about to use this thing a key.